File: src/main/dsa/Arrays/consecutives/ConsecutiveOnes.java
package Arrays.consecutives;

/**
 * Given a binary array nums, return the maximum number of consecutive 1's in the array.
 *  Max Consecutive Ones
 *
 *
 *  Example 1:
 *
 * Input: nums = [1,1,0,1,1,1]
 * Output: 3
 * Explanation: The first two digits or the last three digits are consecutive 1s. The maximum number of consecutive 1s is 3.
 * Example 2:
 *
 * Input: nums = [1,0,1,1,0,1]
 * Output: 2
 *
 *
 * Constraints:
 *
 * 1 <= nums.length <= 105
 * nums[i] is either 0 or 1.
 *
 */
public class ConsecutiveOnes {

    public static int findMaxConsecutiveOnes(int[] nums) {

        int max = 0, consecutiveMax = 0;

        for (int item: nums)
            max = Math.max(max, consecutiveMax = item == 0 ? 0 : consecutiveMax + 1);

        return max;
    }



    public static void main(String[] args) {
        // Print consecutive ones
        int[]  numbers = {1,1,0,1,1,1};
        System.out.println(findMaxConsecutiveOnes(numbers)+ " output ");
    }
}

File: src/main/dsa/Arrays/consecutives/ConsecutiveOnes.kt
package Arrays.consecutives

fun main(args: Array<String>) {
    val numbers : IntArray = intArrayOf(1,1,0,1,1,1)

    print("output is => "+getMaxConsecutiveOnes(numbers))
}

fun getMaxConsecutiveOnes(num: IntArray): Int {
    var max = 0;
    var counter = 0;
    num.forEach { item ->
        when (item){
            1 -> counter++
            else -> {
                max = Math.max(counter, max);
                counter = 0
            }
        }
    }

    return Math.max(counter,max);
}